pages/blog/subscriptions-and-live-queries-real-time-with-graphql.mdx

```diff
@@ -0,0 +1,867 @@
+
+```ts
+import { InMemoryLiveQueryStore } from '@n1ru4l/in-memory-live-query-store';
+import { parse } from 'graphql';
+import { schema } from './schema';
+
+const inMemoryLiveQueryStore = new InMemoryLiveQueryStore();
+
+const rootValue = {
+  todos: [
+    {
+      id: '1',
+      content: 'Foo',
+      isComplete: false,
+    },
+  ],
+};
+
+inMemoryLiveQueryStore
+  .execute({
+    schema,
+    operationDocument: parse(/* GraphQL */ `
+      query todosQuery @live {
+        todos {
+          id
+          content
+          isComplete
+        }
+      }
+    `),
+    rootValue: rootValue,
+    contextValue: {},
+    variableValues: null,
+    operationName: 'todosQuery',
+  })
+  .then(async (result) => {
+    if (isAsyncIterable(result)) {
+      for (const value of result) {
+        // Send to client in real-world app :)
+        console.log(value);
+      }
+    }
+  });
+
+// Invalidate by resource identifier
+rootValue.todos[0].isComplete = true;
+inMemoryLiveQueryStore.invalidate(`Todo:1`);
+
+// Invalidate by root query field coordinate
+rootValue.todos.push({ id: '2', content: 'Baz', isComplete: false });
+inMemoryLiveQueryStore.invalidate(`Query.todos`);
+```
+
+If you are using a ORM library you can simply add a middleware for automatically invalidating resource. Here an example with prisma:
+
+```ts
+// Register Middleware for automatic model invalidation
+prisma.$use(async (params, next) => {
+  const resultPromise = next(params);
+
+  if (params.action === 'update') {
+    resultPromise.then((res) => {
+      if (res?.id) {
+        // invalidate `Post:1` on update
+        liveQueryStore.invalidate(`${params.model}:${res.id}`);
+      }
+    });
+  }
+
+  return resultPromise;
+});
+```
+
+Regarding the transport. Both `graphql-ws` (GraphQL over WebSocket), `graphql-helix` (GraphQL over SEE) and `@n1ru4l/socket-io-graphql-server` (GraphQL over Socket.io), support this library out of the box.
+
+[You can find all libraries and a example todo app that sync across all connected clients over here.](https://github.com/n1ru4l/graphql-live-query)
+
+[More information on how the `InMemoryLiveQueryStore` keeps track of resources here.](https://dev.to/n1ru4l/collecting-graphql-live-query-resource-identifier-with-graphql-tools-5fm5)
+
+So let's take a look at this implementation regarding the aspects we used before.
+
+1. Automatically updates execution result. **Yes.** ✅
+
+The approach of pushing the invalidation responsibility to the user might at first seem like a drawback, but a smart abstraction (such as a ORM middleware), can result in pretty responsive applications.
+
+2. Efficiently send updates to client. **No.** ❌
+
+The execution result published by the AsyncIterator is always a complete execution result. But, a library that can be plugged and create JSON patches from the last to the next execution result is already prototyped and planned to be released soon!
```

Comment:
TODO: update with information of https://github.com/n1ru4l/graphql-live-query/pull/243